### PR TITLE
[APR-230] Add a new destination to support sending events/service checks to the Datadog platform

### PR DESCRIPTION
**Pull request by rayz**
# Context
In order to support sending events and service checks to Datadog, we need a new destination to send them to.

# Solution
We create a new destination in `saluki-components/src/destinations/datadog_events_service_checks` that supports both events and service checks.

# Testing
Submit events and service checks. You should now see them show up on the Datadog platform.

Finishes #120

### CODE REVIEW

File: lib/saluki-components/src/sources/dogstatsd/mod.rs [hunk 1]
```diff
@@ -453,7 +453,7 @@ async fn drive_stream(
                 let maybe_service_checks_event_buffer = match event_buffer.has_data_type(DataType::ServiceCheck) {
                     true => {
                         let mut service_check_event_buffer = source_context.event_buffer_pool().acquire().await;
-                        service_check_event_buffer.extend(event_buffer.extract(Event::is_eventd));
+                        service_check_event_buffer.extend(event_buffer.extract(Event::is_service_check));
```
Comment (rayz) [inline lib/saluki-components/src/sources/dogstatsd/mod.rs:456]: 😅

File: lib/saluki-components/src/destinations/datadog_events_service_checks/mod.rs [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        Ok(Box::new(DatadogEventsServiceChecks {
+            http_client,
+            events_request_builder,
+            service_checks_request_builder,
+        }))
+    }
+}
+
+impl MemoryBounds for DatadogEventsServiceChecksConfiguration {
+    fn specify_bounds(&self, _builder: &mut MemoryBoundsBuilder) {}
```
Comment (rayz) [inline lib/saluki-components/src/destinations/datadog_events_service_checks/mod.rs:122]: Am not sure how to come up with the bounds.

File: lib/saluki-components/src/destinations/datadog_events_service_checks/mod.rs [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+                                        }
+                                    }
+                                    Err(e) => {
+                                        error!(error = %e, "Failed to create request for service check.");
+                                        continue;
+                                    }
+                                }
+                            }
+                        }
+                        _ => {}
```
Comment (tobz) [inline lib/saluki-components/src/destinations/datadog_events_service_checks/mod.rs:194]: We should emit an error message here if we're getting events that aren't events or service checks.

**Review by tobz — CHANGES_REQUESTED** [review]

File: lib/saluki-event/src/eventd/mod.rs [hunk 4]
```diff
@@ -57,6 +68,16 @@ impl AlertType {
             _ => Some(AlertType::Info),
         }
     }
+
+    /// Returns stringified name of the AlertType variant.
+    pub fn stringify(self) -> &'static str {
```
Comment (tobz) [inline lib/saluki-event/src/eventd/mod.rs:73]: This should be `as_str`.

File: lib/saluki-event/src/eventd/mod.rs [hunk 5]
```diff
@@ -89,10 +119,18 @@ impl Priority {
             _ => Some(Priority::Normal),
         }
     }
+
+    /// Returns stringified name of the Priority variant.
+    pub fn stringify(self) -> &'static str {
```
Comment (tobz) [inline lib/saluki-event/src/eventd/mod.rs:124]: This should be `as_str`.

File: lib/saluki-components/src/destinations/datadog_events_service_checks/request_builder.rs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        api_key: String, api_base_uri: Uri, endpoint: EventsServiceChecksEndpoint,
+    ) -> Result<Self, RequestBuilderError> {
+        let api_uri = build_uri_for_endpoint(api_base_uri, endpoint)?;
+
+        Ok(Self { api_key, api_uri })
+    }
+
+    /// Creates the request to be sent by the Http Client.
+    pub fn create_request(&self, data: String) -> Result<Request<String>, RequestBuilderError> {
+        // let write_buffer = self.buffer_pool.acquire().await;
```
Comment (tobz) [inline lib/saluki-components/src/destinations/datadog_events_service_checks/request_builder.rs:32]: Seems like this line should get removed?

File: lib/saluki-components/src/destinations/datadog_events_service_checks/mod.rs [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+        let Self {
+            http_client,
+            mut events_request_builder,
+            mut service_checks_request_builder,
+        } = *self;
+
+        // Spawn our IO task to handle sending requests.
+        let (io_shutdown_tx, io_shutdown_rx) = oneshot::channel();
+        let (requests_tx, requests_rx) = mpsc::channel(32);
+        // let events_service_checks = Metrics::from_component_context(context.component_context());
```
Comment (tobz) [inline lib/saluki-components/src/destinations/datadog_events_service_checks/mod.rs:135]: Seems like this line should get removed?

**Review by tobz — CHANGES_REQUESTED** [review]

File: lib/saluki-components/src/destinations/datadog_events_service_checks/mod.rs (hunk 7 above)
Comment (rayz) [reply lib/saluki-components/src/destinations/datadog_events_service_checks/mod.rs:135]: Oops, good catch

File: lib/saluki-event/src/eventd/mod.rs (hunk 5 above)
Comment (rayz) [reply lib/saluki-event/src/eventd/mod.rs]: I took out the doc since the `fn` doesn't actually need to be `pub` but lmk if you want it back in lol

**Review by tobz — APPROVED** [review]

**Pull request merged**